src/Form/DataTransformerResolver.php

```diff
@@ -77,7 +77,7 @@ public function resolve(
         if (FieldDescriptionInterface::TYPE_CHOICE === $fieldType) {
             $className = $fieldDescription->getOption('class');
```

Comment:
```suggestion
            /** @var class-string|null $className */
            $className = $fieldDescription->getOption('class');
```
Could be a way to fix this.

Reply:
But maybe it's better to check something like `class_exists($className)`

Reply:
Added the class_exists, check if method getTargetModel is not equal to NULL and the Psalm definition. Still get errors related to PHPStan and Psalm, but I don't think these are related to my changes?

Reply:
I will look at it after the merge of https://github.com/sonata-project/SonataAdminBundle/pull/7716 which fix already multiple psalm/phpstan issue

Reply:
Thanks!